### PR TITLE
Fix input-text cursor jump react 18

### PR DESCRIPTION
**Pull request by MKaHead**
Fix input-text cursor jump react 18 #2373 

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch-webhub-cloud/pattern-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: src/components/10-atoms/input-text/index.js [hunk 1]
```diff
@@ -215,13 +216,19 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
   }
 
   handleInput(ev) {
+    this.cursor = ev.target.selectionStart;
```
Comment (MarekLacoAXA) [inline src/components/10-atoms/input-text/index.js]: I suggest to reuse the existing naming: selectionStart.
(maybe alternative: targetSelectionStart)

```suggestion
    this.selectionStart = ev.target.selectionStart;
```

**Review by MarekLacoAXA — COMMENTED** [review]
Thanks for fixing this issue! 👍

File: src/components/10-atoms/input-text/index.js (hunk 1 above)
Comment (MKaHead) [reply src/components/10-atoms/input-text/index.js]: Thank you, the variable can also be function scoped.

File: src/components/10-atoms/input-text/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ## 6.3.0
 
-Add boolean attribute `autocomplete` to put the HTML autocomplete attribute to "on". The default is "off" 
+Feat: Add boolean attribute `autocomplete` to put the HTML autocomplete attribute to "on". The default is "off" #2364
```
Comment (markus-walther) [inline src/components/10-atoms/input-text/CHANGELOG.md:3]: Interesting - but where is the implementation?

File: src/components/10-atoms/input-text/index.js [hunk 3]
```diff
@@ -216,13 +216,19 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
...
     // sets this.invalidFormat
     this._formatCurrency(ev.target.value);
     this.onChange(ev, this.invalidFormat);
     // are we a 'controlled' input in the React sense?
     if (this.isControlled) {
       // yes, set UI from model state
       this.nativeInput.value = this.modelValue;
+
+      requestAnimationFrame(() => {
+        this.nativeInput.setSelectionRange(selectionStart, selectionStart);
```
Comment (markus-walther) [inline src/components/10-atoms/input-text/index.js:230]: OK, this is a creative solution, that seems to be usable in all browsers we care about - for the narrow case of React controlled components with a trivial selection. Just curious: what if the selection is a non-trivial range spanning a few letters? Is it acceptable that the range is lost here? (I genuinely don't know).

**Review by markus-walther — CHANGES_REQUESTED** [review]
Hi Marco,

two questions remain, otherwise good work!

Cheers, Markus

File: src/components/10-atoms/input-text/CHANGELOG.md (hunk 2 above)
Comment (MKaHead) [reply src/components/10-atoms/input-text/CHANGELOG.md:3]: I have develop rebased, feature autocomplete is already in develop and affects the same component.

File: src/components/10-atoms/input-text/index.js (hunk 3 above)
Comment (MKaHead) [reply src/components/10-atoms/input-text/index.js:230]: We would have to find out. That's why I wrote Marek directly that we have to observe the component. I think at the moment is the best fix.

**Comment by MarekLacoAXA** [comment]
Do you have an explanation why does this issue happen? What is different in React-18 compared to React-17? 
I would like to understand what is the root cause of this.

File: src/components/10-atoms/input-text/CHANGELOG.md (hunk 2 above)
Comment (markus-walther) [reply src/components/10-atoms/input-text/CHANGELOG.md:3]: I see, good.

File: src/components/10-atoms/input-text/index.js (hunk 3 above)
Comment (markus-walther) [reply src/components/10-atoms/input-text/index.js:230]: Ok, then good for now.

**Review by markus-walther — APPROVED** [review]
Hi Marco,

LGTM.

--M.

**Comment by MarekLacoAXA** [comment]
Thanks for fixing Marco!
Works quite good, in 99% cases.
(However I was still able to experience this cursor jump "sometimes": in 1% of keypresses or so.)

**Comment by MarekLacoAXA** [comment]
Still reproducible when "typing quicky": can by simulated by pressing several (two/three) keys at once repeatedly on the begin of text. Cursor jumps to end.

**Pull request merged**